packages/teams-js/src/public/pages.ts

```diff
@@ -399,10 +401,17 @@ export namespace pages {
         throw errorNotSupportedOnPlatform;
       }
       const apiVersionTag: string = getApiVersionTag(pagesTelemetryVersionNumber, ApiName.Pages_NavigateToApp);
+
       if (runtime.isLegacyTeams) {
-        resolve(sendAndHandleStatusAndReason(apiVersionTag, 'executeDeepLink', createTeamsAppLink(params)));
+        const typeSafeParameters: AppNavigationParameters = !isAppNavigationParametersObject(params)
+          ? convertNavigateToAppParamsToAppNavigationParameters(params)
```

Comment:
This is nitpicky, but I think if `convertNavigateToAppParamsToAppNavigationParameters` (lol that's a mouthful) was changed to accept both types and then did the `!isAppNavigationParametersObject` check internally, you could write code here that looked like:

```typescript
const typeSafeParameters: AppNavigationParameters = upgradeNavigateToAppParamsIfNeeded(params);
```

Reply:
I understand your feedback and I do like seeing easy-to-read code. My concern with this approach is that I generally don't prefer functions which "might or might not" do a thing because it makes it difficult for the caller to predict what will happen or see what did happen. If I were going to do it, I'd probably leave the `convertNavigateToAppParamsToAppNavigationParameters` function as it is and then create your suggested function as a wrapper (so that we still had a predictable atomic function).

Given that you said it was nitpicky and that you also said (separately) that I could take or leave, I think I will choose to leave this one. Thank you for your suggestion.

Oh, I would also consider your suggestion further if there were multiple places in the code where I was doing this check but since this seems to be the only one, I'll leave it.